contractsV5/KyberNetwork.sol

```diff
@@ -893,9 +930,16 @@ contract KyberNetwork is Withdrawable, Utils, IKyberNetwork, ReentrancyGuard {
     }
     
     function parseTradeDataHint(TradeData memory tradeData,  bytes memory hint) internal view {
-        tradeData.tokenToEth.addresses = reservesPerTokenSrc[address(tradeData.input.src)];
-        tradeData.ethToToken.addresses = reservesPerTokenDest[address(tradeData.input.dest)];
-        //PERM_HINT is treated as no hint, so we just return
-        if (hint.length == 0 || keccak256(hint) == keccak256(PERM_HINT)) return;
+        tradeData.tokenToEth.addresses = (tradeData.input.src == ETH_TOKEN_ADDRESS) ? new IKyberReserve[](1) : reservesPerTokenSrc[address(tradeData.input.src)];
+        tradeData.ethToToken.addresses = (tradeData.input.dest == ETH_TOKEN_ADDRESS) ? new IKyberReserve[](1) :reservesPerTokenDest[address(tradeData.input.dest)];
+
+        //PERM is treated as no hint, so we just return
+        if (hint.length == 0 || keccak256(hint) == keccak256("PERM")) {
```

Comment:
for gas consumption
is it better to check perm only of length is 4.
I guess it should be better. please check.
so we don't run two keccak operations. per trade when not required.

Reply:
It's 400 gas more to use keccak.